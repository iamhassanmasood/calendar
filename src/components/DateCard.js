import React, { useState } from "react";
import {
  CardWrapper,
  MonthWrapper,
  MonthHeading,
  DateHeading,
  DaysHeading,
} from "./styledcomponents";

export default function DateCard({ month, date, day, selected }) {
  const [value, setValue] = useState("");

  return (
    <div className="container">
      <div className="card-wrapper" onClick={() => setValue(selected)}>
        <div className="month-wrapper">
          <div className="month-heading">{month}</div>
        </div>
        <div className="date-heading">{date}</div>
        <div className="days-heading">{day}</div>
      </div>
    </div>
  );
}
