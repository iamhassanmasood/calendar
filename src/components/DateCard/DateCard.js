import React from "react";
import { Card, Typography, Col } from "antd";
const { Title, Paragraph } = Typography;

export default function DateCard(props) {
  const {
    month,
    date,
    day,
    onChange,
    disabled,
    slectedDate,
    headStyles,
    bodyStyles,
  } = props;
  return (
    <Col span={24}>
      {!disabled ? (
        <Card
          className="card-enable"
          title={month}
          headStyle={headStyles}
          bodyStyle={bodyStyles}
          onClick={() => onChange(slectedDate)}
        >
          <Title className="date-title-head">{date}</Title>
          <Paragraph className="day-name">
            {disabled ? "Closed" : day}
          </Paragraph>
        </Card>
      ) : (
        <Card
          className={!disabled ? "card-enable" : "card-disable"}
          title={month}
          bodyStyle={bodyStyles}
          headStyle={headStyles}
        >
          <Title className="date-title-head">{date}</Title>
          <Paragraph className="day-name">
            {disabled ? "Closed" : day}
          </Paragraph>
        </Card>
      )}
    </Col>
  );
}
