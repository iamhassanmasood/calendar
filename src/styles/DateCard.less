@mobile: ~"only screen and (min-width: 120px) and (max-width: 368px)";

.card-enable {
  width: 170px;
  height: 180px;
  text-align: center;
  cursor: pointer;
}
.card-disable {
  width: 170px;
  height: 180px;
  text-align: center;
  cursor: no-drop;
}

.date-title-head {
  font-weight: normal !important;
  color: gray !important;
}
.day-name {
  color: gray;
  font-size: 20px;
  bottom: 0;
}

.ant-card {
  border-radius: 10px 10px 0 0;
}
.ant-card-head {
  background: #dd3c3c;
  border-radius: 9px 9px 0 0;
  font-size: 18px;
}
.ant-card-head-title {
  color: #fff;
}

@media @mobile {
  .card-enable {
    width: 110px;
    height: 140px;
    text-align: center;
    cursor: pointer;
  }
  .card-disable {
    width: 110px;
    height: 140px;
    text-align: center;
    cursor: no-drop;
  }
  .day-name {
    color: gray;
    font-size: 10px;
    bottom: 0;
  }
  .date-title-head {
    font-weight: normal !important;
    color: gray !important;
    font-size: 20px !important;
  }
  .ant-card-head {
    background: #dd3c3c;
    border-radius: 9px 9px 0 0;
  }
}
