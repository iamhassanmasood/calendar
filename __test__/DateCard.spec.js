import React from "react";
import { shallow } from "enzyme/build";
import DateCard from "./../src/components/DateCard/DateCard";

describe("date card component", () => {
  it("mounts without crashing", () => {
    const wrapper = shallow(<DateCard />);
    wrapper.unmount();
  });

  it("check dom tree rendering", () => {
    let tree = shallow(<DateCard />);
    expect(tree).toMatchSnapshot();
  });
});
