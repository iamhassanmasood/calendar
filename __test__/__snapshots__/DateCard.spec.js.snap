// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`date card component check dom tree rendering 1`] = `ShallowWrapper {}`;
